modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -63,73 +124,209 @@ def _unstrip_protocol(protocol, path):
                     return path
             return f"{protos[0]}://{path}"
 
-        if isinstance(path, AbstractBufferedFile):
-            return path.fs, [path]
-        fs = get_filesystem_class(split_protocol(path)[0])(**storage_options)
-        dataset = cls._get_dataset(path, storage_options)
+        if isinstance(self.path, AbstractBufferedFile):
+            return self.path.fs, [self.path]
+        fs = get_filesystem_class(split_protocol(self.path)[0])(**self.storage_options)
         # version.parse() is expensive, so we can split this into two separate loops
         if version.parse(fsspec.__version__) < version.parse("2022.5.0"):
-            files = [_unstrip_protocol(fs.protocol, fpath) for fpath in dataset.files]
+            files = [_unstrip_protocol(fs.protocol, fpath) for fpath in files]
         else:
-            files = [fs.unstrip_protocol(fpath) for fpath in dataset.files]
+            files = [fs.unstrip_protocol(fpath) for fpath in files]
 
         return fs, files
 
-    @classmethod
-    def _get_dataset(cls, path, storage_options):
-        """
-        Retrieve parquet dataset for given path.
 
-        Parameters
-        ----------
-        path : str, path object or file-like object
-            Path to dataset.
-        storage_options : dict
-            Parameters for specific storage engine.
+@_inherit_docstrings(Dataset)
+class PyArrowDataset(Dataset):
+    def __init__(self, path, storage_options):
+        self.path = path
+        self.storage_options = storage_options
+        self.dataset = self._init_dataset()
+        self._row_groups_per_file = None
+        self.filesystem, self._files = self._get_files(self.dataset.files)
 
-        Returns
-        -------
-        dataset : ParquetDataset
-            Dataset object of given Parquet file-path.
-        """
+    def _init_dataset(self):  # noqa: GL08
         from pyarrow.parquet import ParquetDataset
 
-        fs, fs_path = cls._get_fs_and_fs_path(path, storage_options)
+        fs, fs_path = self._get_fs_and_fs_path()
         dataset = ParquetDataset(fs_path, filesystem=fs, use_legacy_dataset=False)
         return dataset
 
+    @property
+    def pandas_metadata(self):
+        return self.dataset.schema.pandas_metadata
+
+    @property
+    def columns(self):
+        return self.dataset.schema.names
+
+    @property
+    def engine(self):
+        return "pyarrow"
+
+    def row_groups_per_file(self):
```

Comment:
Per our discussion on slack, seems like our doc checker is breaking right now because of @cache_readonly properties. We should address that issue in another PR. For now, I've added a TODO so that we can come back to this later.